### PR TITLE
Adjust for composer-require-checker check.

### PR DESCRIPTION
**Pull request by sasezaki**
_To avoid conflict, I do PR into master branch_

* Added Composer\Audoload\ClassLoader into `bin/composer-require-checker-config.json` , it is soft dependency symbol.
* DOMElement instead of DomElement - https://php.net/DOMElement
  to avoid composer-require-check reports false-positive,
    just did - `find src/ -type f -exec sed -i -e 's/DomElement/DOMElement/g' {} \;`

### Before
```
$ composer-require-checker --config-file=$(pwd)/bin/composer-require-checker-config.json
ComposerRequireChecker 3.8.0@537138b833ab0f9ad72b667a72bece2a765e88ab
The following 2 unknown symbols were found:
+-------------------------------+--------------------+
| Unknown Symbol                | Guessed Dependency |
+-------------------------------+--------------------+
| Composer\Autoload\ClassLoader |                    |
| DomElement                    |                    |
+-------------------------------+--------------------+
```

### After
```
composer-require-checker --config-file=$(pwd)/bin/composer-require-checker-config.json
ComposerRequireChecker 3.8.0@537138b833ab0f9ad72b667a72bece2a765e88ab
There were no unknown symbols found.
```
* Also composer-require-checker v4 check passed.

### CODE REVIEW

File: src/Psalm/Config.php [hunk 1]
```diff
@@ -764,7 +764,7 @@ private static function validateXmlConfig(string $base_dir, string $file_content
 
         $psalm_nodes = $dom_document->getElementsByTagName('psalm');
 
-        /** @var DomElement|null */
+        /** @var DOMElement|null */
         $psalm_node = $psalm_nodes->item(0);
```
Comment (sasezaki) [inline src/Psalm/Config.php:768]: oh, good catch! 😄

Comment (sasezaki) [reply src/Psalm/Config.php:767]: removed at 04ea5c0

Comment (sasezaki) [reply src/Psalm/Config.php:767]: > https://github.com/vimeo/psalm/runs/5150670385?check_suite_focus=true

I'm currently wondering about this psalm report.

>  Error: src/Psalm/Config.php:767:18: UnnecessaryVarAnnotation: The @var DOMElement|null annotation for $psalm_node is unnecessary (see https://psalm.dev/212)

https://psalm.dev/r/7bd042f521

In this case, inline phpdoc var should be set to suppress error.

Comment (sasezaki) [reply src/Psalm/Config.php:768]: > removed at https://github.com/vimeo/psalm/commit/04ea5c01c74690ccf9585c637e819b50ae0900a7

I did reset this,  just now.

Comment (sasezaki) [reply src/Psalm/Config.php:768]: related : https://github.com/vimeo/psalm/issues/7567

Comment (sasezaki) [reply src/Psalm/Config.php:768]: so, `getElementsByTagName()` returns `DOMNodeList<DOMElement>` is intended behavior.
https://github.com/vimeo/psalm/blob/11e60fa261c682627e6b85300dd6f424b87520b1/stubs/extensions/dom.phpstub#L9

Ok, I would remove this `/** @var DOMElement|null */`

**Comment by AndrolGenhald** [comment]
Can we add this check to composer.json scripts and CI?

**Comment by orklah** [comment]
Thanks!

**Pull request merged**